codegen/templates/client_fixture_types.tmpl

```diff
@@ -0,0 +1,71 @@
+{{- /* template to render client fixture type code */ -}}
+{{- $instance := .Instance }}
+{{- $scenariosMap := .Fixture.Scenarios }}
+package clientmock
+
+import (
+	"context"
+
+	{{range $idx, $pkg := .IncludedPackages -}}
+	{{$pkg.AliasName}} "{{$pkg.PackageName}}"
+	{{end}}
+)
+
+{{- $clientID := .ClientID -}}
+{{$exposedMethods := .ExposedMethods}}
+
+// ClientFixture defines the client fixture type for {{$clientID}}
+type ClientFixture struct {
+{{- range $svc := .Services -}}
+{{range .Methods}}
+{{$serviceMethod := printf "%s::%s" $svc.Name .Name -}}
+{{$methodName := (title (index $exposedMethods $serviceMethod)) -}}
+{{- if (index $scenariosMap $methodName) -}}
+{{$methodName}} *{{$methodName}}Scenarios
+{{- end -}}
+{{- end -}}
+{{- end -}}
+}
+
+{{range $methodName, $scenarios := $scenariosMap -}}
+{{$methodName := pascal $methodName -}}
+// {{$methodName}}Scenarios defines all fixture scenarios for {{$methodName}}
+type {{$methodName}}Scenarios struct {
+{{range $scenario := $scenarios -}}
+{{pascal $scenario}} *{{$methodName}}Fixture `scenario:"{{$scenario}}"`
```

Comment:
I don't see that the tag is used anywhere. What is this for or possibly to be used for?

Reply:
This is intended read by people. It links the struct field with the string literal in `client-config.json`, one can define such scenario
```
{
    "scenarios": [
         "not-found"
     ]
}
```
where "not-found" would be converted to "NotFound" to qualify as a struct field, the tag points to the scenario in above json.